services/order.py

```diff
@@ -0,0 +1,37 @@
+from datetime import datetime
+from django.contrib.auth import get_user_model
+from django.db import transaction
+from django.db.models import QuerySet
+from db.models import Ticket, Order, MovieSession
+
+
+@transaction.atomic
+def create_order(
+    tickets: list[dict],
+    username: str,
+    date: datetime = None,
+) -> None:
+
+    order = Order.objects.create(
+        user=get_user_model().objects.get(username=username)
+    )
+
+    for ticket in tickets:
+        if date:
+            order.created_at = date
+            order.save()

```

Comment:
The `order.save()` call should be moved outside the loop to avoid multiple saves during the creation of tickets. This will optimize the database operations by reducing unnecessary writes.